virtual_rainforest/core/utils.py

```diff
@@ -119,3 +121,49 @@ def set_layer_roles(canopy_layers: int, soil_layers: int) -> List[str]:
         + ["soil"] * soil_layers
     )
     return layer_roles
+
+
+def check_constants(config: dict[str, Any], model_name: str, class_name: str) -> None:
+    """Check that the constants defined in the config are expected.
+
```

Comment:
It is not clear to me where the defaults are stored and how this function checks the inputs, is there a list of allowed variables names somewhere?

Reply:
The defaults are stored within the constants class, e.g. `AbioticSimpleConsts` in `abiotic_simple`'s `constants.py` module. It's not really documented anywhere that this is the case though, so might be worth me making a new documentation page for this.

One thing this function does do is print the full list of possible names which should hopefully make it obvious for the user where they have gone wrong